From 282114078f35aa6ef7e5a71a3875f052ea9217f4 Mon Sep 17 00:00:00 2001
From: Florian Zaruba <florian@openhwgroup.org>
Date: Sun, 11 Apr 2021 23:33:51 +0200
Subject: [PATCH 1/2] Lint with Verilator

Signed-off-by: <florian@openhwgroup.org>
---
 rtl/cv32e40p_cs_registers.sv | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/rtl/cv32e40p_cs_registers.sv b/rtl/cv32e40p_cs_registers.sv
index 4347d27..2c3fbc7 100644
--- a/rtl/cv32e40p_cs_registers.sv
+++ b/rtl/cv32e40p_cs_registers.sv
@@ -1502,7 +1502,7 @@ end //PULP_SECURE
       if( (cnt_gidx == 1) ||
           (cnt_gidx >= (NUM_MHPMCOUNTERS+3) ) )
         begin : gen_non_implemented
-        assign mhpmcounter_q[cnt_gidx] = 'b0;
+        always_ff @(posedge clk) mhpmcounter_q[cnt_gidx] <= 'b0;
       end
       else begin : gen_implemented
         always_ff @(posedge clk, negedge rst_n)
@@ -1533,11 +1533,11 @@ end //PULP_SECURE
       if( (evt_gidx < 3) ||
           (evt_gidx >= (NUM_MHPMCOUNTERS+3) ) )
         begin : gen_non_implemented
-        assign mhpmevent_q[evt_gidx] = 'b0;
+        always_ff @(posedge clk) mhpmevent_q[evt_gidx] <= 'b0;
       end
       else begin : gen_implemented
         if (NUM_HPM_EVENTS < 32) begin : gen_tie_off
-             assign mhpmevent_q[evt_gidx][31:NUM_HPM_EVENTS] = 'b0;
+            always_ff @(posedge clk) mhpmevent_q[evt_gidx][31:NUM_HPM_EVENTS] <= 'b0;
         end
         always_ff @(posedge clk, negedge rst_n)
             if (!rst_n)
@@ -1556,7 +1556,7 @@ end //PULP_SECURE
           (en_gidx == 1) ||
           (en_gidx >= (NUM_MHPMCOUNTERS+3) ) )
         begin : gen_non_implemented
-        assign mcounteren_q[en_gidx] = 'b0;
+        always_ff @(posedge clk) mcounteren_q[en_gidx] <= 'b0;
       end
       else begin : gen_implemented
         always_ff @(posedge clk, negedge rst_n)
@@ -1576,7 +1576,7 @@ end //PULP_SECURE
       if( (inh_gidx == 1) ||
           (inh_gidx >= (NUM_MHPMCOUNTERS+3) ) )
         begin : gen_non_implemented
-        assign mcountinhibit_q[inh_gidx] = 'b0;
+        always_ff @(posedge clk) mcountinhibit_q[inh_gidx] <= 'b0;
       end
       else begin : gen_implemented
         always_ff @(posedge clk, negedge rst_n)
-- 
2.25.1.377.g2d2118b814

